src/types/from_sql.rs

```diff
@@ -15,6 +15,10 @@ pub enum FromSqlError {
     /// requested type.
     OutOfRange(i64),
 
+    /// Error when the blob result returned by SQLite cannot be stored into the
+    /// requested type due to a size mismatch. (type size, blob size)
+    InvalidSize(usize, usize),
```

Comment:
As mentioned, I think this should be something like `InvalidBlobSize` (perhaps even naming the fields? They're kind of opaque).